### DISCUSSION TITLE
How would I `preload` stylesheets imported in layouts?

### DISCUSSION THREAD

**Discussion by flexdinesh** [withastro/roadmap#421] [Proposal]
Preloading styles to improve render performance: [Link types: preload](https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/preload)

Currently if I import a style in my layout, the HTML output adds it as a link stylesheet.
For example,
```
// Layout.astro
import "../styles/global.css";
```
is output as 
```
<link rel="stylesheet" href="/assets/_slug_.b0fc1e6b.css">
```

I would like to provide hints to Astro that the stylesheet I'm importing is a critical stylesheet that I would like to preload as style.

```
<link rel="preload" href="/assets/_slug_.b0fc1e6b.css" as="style" />
<link rel="stylesheet" href="/assets/_slug_.b0fc1e6b.css" />
```

Is there a plugin or config currently out there I could use to achieve this?

**Comment by florian-lefebvre** [comment]
@delucis I don't think we have this yet so do you know if preloading is actually important for styles like this?

**Reply by delucis** [reply]
I guess you can achieve it manually in a head component:

```astro
---
import styleUrl from '../styles/global.css?url&no-inline';
---

<link rel="preload" href={styleUrl} as="style">
<link rel="stylesheet" href={styleUrl}>
```

BUT as you said, I don’t think this specific usage would make sense — browsers already discover a `<link>` tag like this with the preload scanner and don’t need an explicit `rel="preload"` link.

The scenarios where preloading styles makes sense would be where you’re not doing `import '../styles/global.css'` in a component. The things that come to mind:

- A third-party style included via `@import` _inside_ CSS. This would only be discoverable by a browser once the CSS is loaded, so adding a preload link could improve loading. (You should never include a third-party style via `@import` though 😁)

- You have a component that dynamically injects style links based on clientside data or interaction. (Also probably not something you should do unless your CSS is reallllly huge, but then you have other problems 😅)

But yeah, I don’t see a scenario where you `import` a CSS file in Astro, Astro adds `<link>` for you, and you would ALSO need a preload.
